modules/linear_boltzmann_solvers/a_lbs_solver/lbs_solver.cc

```diff
@@ -2100,180 +2100,329 @@ LBSSolver::ReadRestartData(const std::string& folder_name, const std::string& fi
 
-  char header_bytes[320];
-  memset(header_bytes, '-', 320);
-  strncpy(header_bytes, header_info.c_str(), std::min(header_size, 319));
-  header_bytes[319] = '\0';
+  char header_bytes[num_bytes];
+  memset(header_bytes, '-', num_bytes);
+  strncpy(header_bytes, header_info.c_str(), std::min(header_size, num_bytes - 1));
+  header_bytes[num_bytes - 1] = '\0';
 
   file << header_bytes;
 
-  // Get relevant items
-  auto NODES_ONLY = chi_math::UnknownManager::GetUnitaryUnknownManager();
+  // Write macro info
+  const auto& uk_man = groupset.psi_uk_man_;
+  const auto NODES_ONLY = chi_math::UnknownManager::GetUnitaryUnknownManager();
 
-  size_t num_local_nodes = discretization_->GetNumLocalDOFs(NODES_ONLY);
-  size_t num_angles = groupset.quadrature_->abscissae_.size();
-  size_t num_groups = groupset.groups_.size();
-  size_t num_local_dofs = psi_new_local_[groupset.id_].size();
-  auto dof_handler = groupset.psi_uk_man_;
+  const uint64_t num_local_nodes = discretization_->GetNumLocalDOFs(NODES_ONLY);
+  const unsigned int num_gs_angles = groupset.quadrature_->abscissae_.size();
+  const unsigned int num_gs_groups = groupset.groups_.size();
+  const auto num_local_gs_dofs = discretization_->GetNumLocalDOFs(uk_man);
 
-  // Write num_ quantities
-  file.write((char*)&num_local_nodes, sizeof(size_t));
-  file.write((char*)&num_angles, sizeof(size_t));
-  file.write((char*)&num_groups, sizeof(size_t));
-  file.write((char*)&num_local_dofs, sizeof(size_t));
+  // clang-format off
+  ChiLogicalErrorIf(src.size() != num_local_gs_dofs,
+                    "Incompatible angular flux vector provided for groupset " +
+                    std::to_string(groupset.id_) + ".");
+  // clang-format on
 
-  auto& sdm = discretization_;
+  file.write((char*)&num_local_nodes, sizeof(uint64_t));
+  file.write((char*)&num_gs_angles, sizeof(unsigned int));
+  file.write((char*)&num_gs_groups, sizeof(unsigned int));
 
-  // Write per dof data
-  size_t dof_count = 0;
+  // Write the groupset angular flux data
   for (const auto& cell : grid_ptr_->local_cells)
-  {
-    const size_t num_nodes = sdm->GetCellNumNodes(cell);
-    for (unsigned int i = 0; i < num_nodes; ++i)
-      for (unsigned int n = 0; n < num_angles; ++n)
-        for (unsigned int g = 0; g < num_groups; ++g)
+    for (unsigned int i = 0; i < discretization_->GetCellNumNodes(cell); ++i)
+      for (unsigned int n = 0; n < num_gs_angles; ++n)
+        for (unsigned int g = 0; g < num_gs_groups; ++g)
         {
-          if (++dof_count > num_local_dofs) goto close_file;
-
-          uint64_t dof_map = sdm->MapDOFLocal(cell, i, dof_handler, n, g);
-          double value = psi_new_local_[groupset.id_][dof_map];
+          const uint64_t dof_map = discretization_->MapDOFLocal(cell, i, uk_man, n, g);
+          const double value = src[dof_map];
 
-          file.write((char*)&cell.global_id_, sizeof(size_t));
+          file.write((char*)&cell.global_id_, sizeof(uint64_t));
```

Comment:
what's the rule we follow for size_t versus other denominations of sizes?

Reply:
There is no hard and fast rule used int he code today. We should discuss adopting some rules for this. My thought here to use `uint64_t` is to be as explicit as possible.